text/3373-avoid-nonlocal-definitions-in-fns.md

```diff
@@ -0,0 +1,71 @@
+- Feature Name: N/A
+- Start Date: 2022-01-19
+- RFC PR: [rust-lang/rfcs#3373](https://github.com/rust-lang/rfcs/pull/3373)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Starting in Rust 2024, stop allowing items inside functions or expressions to
+implement methods or traits that are visible outside the function or
+expression.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, tools cross-referencing uses and definitions (such as IDEs) must
+either search inside all function bodies and other expression-containing items
+to find potential definitions corresponding to uses within another function, or
+not cross-reference those definitions at all.
+
+Humans cross-referencing such uses and definitions may find themselves
+similarly baffled.
+
+With this change, both humans and tools can limit the scope of their search and
+avoid looking for definitions inside other functions or items, without missing
+any relevant definitions.
+
+# Explanation
```

Comment:
Was this made off an old template?  It's missing a bunch of the sections from https://github.com/rust-lang/rfcs/blob/master/0000-template.md

(Not a huge deal, but procedurally I think it ought to have everything from the template, even if some of those sections have minimal content.)